nni/experiment/config/common.py

```diff
@@ -106,7 +106,10 @@ def _validation_rules(self):
     'trial_code_directory': util.canonical_path,
     'max_experiment_duration': lambda value: f'{util.parse_time(value)}s' if value is not None else None,
     'experiment_working_directory': util.canonical_path,
-    'tuner_gpu_indices': lambda value: [int(idx) for idx in value.split(',')] if isinstance(value, str) else value
+    'tuner_gpu_indices': lambda value: [int(idx) for idx in value.split(',')] if isinstance(value, str) else value,
+    'tuner': lambda config: None if config.name == '_none_' else config,
+    'assessor': lambda config: None if config.name == '_none_' else config,
+    'advisor': lambda config: None if config.name == '_none_' else config,
```

Comment:
where '_none_' is assigned to `name` and in which case?

Reply:
found...

Reply:
In `Experiment.__init__()`. The purpose is to let users omit calling `AlgorithmConfig()` explicitly.
`AlgorithmConfig` with name "_none_" is designed to be equivalent to `None`, just like `Path('/home')` is equivalent to `str('/home/')`. This means everywhere `tuner` can be `None`, it can also be `AlgorithmConfig(name='_none_')`.